### PR TITLE
Support GCS (Google Cloud Storage) as storage for tfmigrate plan files

### PR DESCRIPTION
**Pull request by KengoTODA**
From [v0.3.7](https://github.com/minamijoyo/tfmigrate/releases/tag/v0.3.7), tfmigrate now [supports GCS as history backend](https://github.com/minamijoyo/tfmigrate/pull/103).

I believe that it is possible for tfaction to realize #59, so here I'm going to propose a change to close it.

To be honest, I'm not so sure how I should test tfaction.
I've confirmed that the added GitHub action can print expected outputs, but please tell me if it is not enough to keep this new feature tested well. :bow:

### CODE REVIEW

**Comment by suzuki-shunsuke** [comment]
Thank you for your contribution!

**Comment by suzuki-shunsuke** [comment]
It looks good.
I'll test.

File: tfmigrate-plan/main.sh [hunk 1]
```diff
@@ -12,9 +12,18 @@ if [ -n "${GCS_BUCKET_NAME_PLAN_FILE:-}" ]; then
 fi
 
 if [ ! -f .tfmigrate.hcl ]; then
-	sed "s|%%TARGET%%|$TFACTION_TARGET|g" \
-		"$GITHUB_ACTION_PATH/tfmigrate.hcl" |
-		sed "s|%%S3_BUCKET_NAME_TFMIGRATE_HISTORY%%|$S3_BUCKET_NAME_TFMIGRATE_HISTORY|g" >.tfmigrate.hcl
+	if [ -n "${S3_BUCKET_NAME_PLAN_FILE:-}" ]; then
```
Comment (suzuki-shunsuke) [inline tfmigrate-plan/main.sh:15]:
```suggestion
	if [ -n "${S3_BUCKET_NAME_TFMIGRATE_HISTORY:-}" ]; then
```

File: tfmigrate-plan/main.sh [hunk 2]
```diff
@@ -12,9 +12,18 @@ if [ -n "${GCS_BUCKET_NAME_PLAN_FILE:-}" ]; then
...
+		sed "s|%%TARGET%%|$TFACTION_TARGET|g" \
+			"$GITHUB_ACTION_PATH/tfmigrate.hcl" |
+			sed "s|%%S3_BUCKET_NAME_TFMIGRATE_HISTORY%%|$S3_BUCKET_NAME_TFMIGRATE_HISTORY|g" >.tfmigrate.hcl
+	elif [ -n "${GCS_BUCKET_NAME_PLAN_FILE:-}" ]; then
```
Comment (suzuki-shunsuke) [inline tfmigrate-plan/main.sh:19]:
```suggestion
	elif [ -n "${GCS_BUCKET_NAME_TFMIGRATE_HISTORY:-}" ]; then
```

**Comment by suzuki-shunsuke** [comment]
I've confirmed it works well.
I'm updating document.

**Comment by KengoTODA** [comment]
Thanks for your review and documentation!

**Comment by suzuki-shunsuke** [comment]
Published the prerelease version https://github.com/suzuki-shunsuke/tfaction/releases/tag/v0.5.16-0
I'll update the document and release v0.5.16.

**Comment by suzuki-shunsuke** [comment]
https://github.com/suzuki-shunsuke/tfaction-docs/pull/57
https://github.com/suzuki-shunsuke/tfaction/pull/481

**Comment by suzuki-shunsuke** [comment]
Released. https://github.com/suzuki-shunsuke/tfaction/releases/tag/v0.5.16

**Pull request merged**